primer-api/primer-api.cabal

```diff
@@ -0,0 +1,199 @@
+cabal-version: 3.0
+name:          primer-api
+version:       0.7.2.0
+license:       AGPL-3.0-or-later
+license-file:  COPYING
+copyright:     (c) 2023 Hackworth Ltd
+maintainer:    src@hackworthltd.com
+author:        Hackworth Ltd <src@hackworthltd.com>
+stability:     experimental
+synopsis:      An API for Primer
+category:      API
+
+library
+  exposed-modules:
+    Primer.API
+    Primer.API.NodeFlavor
+    Primer.API.RecordPair
+
+  hs-source-dirs:     src
+  default-language:   GHC2021
+  default-extensions:
+    NoImplicitPrelude
+    DataKinds
+    DeriveAnyClass
+    DerivingStrategies
+    DerivingVia
+    LambdaCase
+    OverloadedStrings
+
+  ghc-options:
+    -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates
+    -Wcompat -Widentities -Wredundant-constraints
+    -Wmissing-deriving-strategies -fhide-source-paths
+
+  build-depends:
+    , aeson                        >=2.0      && <2.2
```

Comment:
I wonder if any build-depends can be trimmed, either here or in the primer package now the api has been split?

Reply:
Yes, thanks, I meant to revisit that and forgot before I pushed this PR.